addons/website/static/src/js/utils.js

```diff
@@ -239,6 +239,32 @@ function websiteDomain(self) {
     return ['|', ['website_id', '=', false], ['website_id', '=', websiteID]];
 }
 
+/**
+ * Checks if the 2 given URLs originates from the same domain.
+ * It will consider naked URL and `www` URL originating from the same domain.
+ * It will consider `https` URL `http` URL originating from the same domain.
+ *
+ * @param {string} url1
+ * @param {string} url2
+ * @returns {Boolean}
+ */
+function areSameDomain(url1, url2) {
+    try {
+        url1 = new URL(url1).host;
+        url2 = new URL(url2).host;
+    } catch (_err) {
+        // Incorrect URL, `false` URL..
+        return false;
+    }
+    if (url1 === url2) {
+        return true;
+    }
+    if (url1.replace('www.', '') === url2.replace('www.', '')) {
```

Comment:
Would not cost much to write `.replace(/^www./, '')` to be accurate (even if unlikely to break any usecase I guess).

Reply:
I got lazy on this one :grimacing: 
https://github.com/odoo/odoo/compare/3d7c5f19f8690d0efc081c82b9646189806c0b2e..756b4f4aa4ea301d4231ad24b207ee976cdb2540